### PR TITLE
Add tests for client bindings

### LINKED ISSUE #8636: Non-repudiation: allow supported bindings to use non-repudiation
The existing bindings clients do not allow to add the necessary interceptors that add support to submit commands over a non-repudiation proxy. Extend supported bindings to do so and possibly also those used internally (e.g. Scala bindings and client). The most flexible way to do this is probably to allow callers to pass in arbitrary interceptors.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
changelog_end

Closes #8636

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
@cocreature After some thinking I realized that thanks to the fact that our bindings accept a channel builder as a parameter, we can simply add interceptors there, without adding anything explicitly to our bindings. Do you think this should be enough for now? I guess we can add something explicitly for interceptors, if there is a request for it.

**Comment by cocreature** [comment]
@stefanobaghino-da sounds like it’s enough for now, didn’t know you could add interceptors there

**Comment by stefanobaghino-da** [comment]
> @stefanobaghino-da sounds like it’s enough for now, didn’t know you could add interceptors there

Yeah, it completely slipped my mind until I had a better look. 😅

File: runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          CommandService.scalaDescriptor.fullName,
+          CommandSubmissionService.scalaDescriptor.fullName,
+        )
+      } yield proxy
+
+    proxy.use { _ =>
+      val clientChannelBuilder =
+        builders.proxyChannel
+          .asInstanceOf[NettyChannelBuilder]
+          .intercept(new SigningInterceptor(key, certificate))
```
Comment (cocreature) [inline runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:70]: What happens if I use something other than the command service? From a quick look, I couldn’t find anything that prevents the interceptor from kicking in. iirc we only insert it for command services on the non-repudation server side so it probably doesn’t break anything but unnecessarily signing things also doesn’t seem that great of an idea.

At the very least, I think we should have a test for accessing other services (e.g., one of acs, transaction service, …). Totally fine to do this in another PR.

**Review by cocreature — APPROVED** [review]
Thanks! One comment on non-command services

File: runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:70]: Very good point, thanks. Probably we actually need something specific.

Comment (stefanobaghino-da) [reply runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:70]: I've addressed this with https://github.com/digital-asset/daml/pull/9050. After that is merged, I can rebase this on `main`, limit signing to commands and add a test for this.

Comment (stefanobaghino-da) [reply runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/9036/commits/dec170bad4fb2d3a790723ea5343d0f3acba08d3

File: runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          .asInstanceOf[NettyChannelBuilder]
+          .intercept(SigningInterceptor.signCommands(key, certificate))
+      val client = DamlLedgerClient.newBuilder(clientChannelBuilder).build()
+      client.connect()
+      val command = ProtoCommand.parseFrom(generateCommand().getCommands.commands.head.toByteArray)
+      val expectedWorkflowId = "workflow-id"
+      val expectedApplicationId = "application-id"
+      val expectedCommandId = "command-id"
+      val expectedParty = "party-1"
+      client.getCommandClient
```
Comment (cocreature) [inline runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:78]: Can we get a test that hits a non-command endpoint to make sure nothing breaks?

Comment (stefanobaghino-da) [reply runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:78]: Oops, sorry, I forgot to push a commit.

Comment (stefanobaghino-da) [reply runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala:78]: https://github.com/digital-asset/daml/pull/9053

**Pull request merged**